escape_roomba/event_logger.py

```diff
@@ -0,0 +1,214 @@
+
+    async def debug_on_reaction_clear_emoji(self, r):
+        self.debug(f'Clear [{r.emoji}] from\n    {self._format(m=r.message)}')
+
+    async def debug_on_raw_reaction_clear_emoji(self, p):
+        self.debug(f'Raw clear [{p.emoji}] from\n    ' +
+                   self._format(g=p.guild_id, c=p.channel_id, m=p.message_id))
+
+    async def debug_on_guild_channel_delete(self, c):
+        self.debug(f'Channel delete {self._format(c=c)}')
+
+    async def debug_on_guild_channel_create(self, c):
+        self.debug(f'Channel create {self._format(c=c)}')
+
+    async def debug_on_guild_channel_update(self, b, a):
+        self.debug(f'Channel update {self._format(c=b)}')
+
+    async def debug_on_guild_available(self, g):
+        self.debug(f'Server available {self._format(g=g)}')
+
+    async def debug_on_guild_unavailable(self, g):
+        self.debug(f'Server UN-available {self._format(g=g)}')
+
+    def _format(self, g=None, c=None, u=None, m=None):
+        """Pretty-print Guild (or ID), (Guild/Private)Channel (or ID),
+        User/Member (or ID), and/or Message (or ID) for some object.
+        Returns a string formatted for debug output."""
+
+        # Look up raw IDs, if possible.
+        if isinstance(g, int):
+            g = self.context.client.get_guild(g) or g
+        if isinstance(c, int):
+            c = self.context.client.get_channel(c) or c
+        if isinstance(u, int):
+            u = self.context.client.get_user(u) or u
+
+        # Get embedded values from objects, if present.
+        if isinstance(m, discord.Message):
+            g = m.guild or g
+            c = m.channel or c
+            u = m.author or u
+        if isinstance(u, discord.Member):
+            g = u.guild or g
+        if isinstance(c, discord.abc.GuildChannel):
+            g = c.guild or g
+
+        # Accumulate output that will be assembled with spaces.
+        # The final format will be part or all of this:
+        #    "guild" #channel (user#1234) "message text"
+        out = []
+
+        if isinstance(g, discord.Guild):
+            out.append(f'"{g}"')
+        elif g:
+            out.append(f'g={format_id(g)}')
+
+        if isinstance(c, discord.abc.GuildChannel):
+            out.append(f'#{c}')
+        elif isinstance(c, discord.abc.PrivateChannel):
+            out.append(f'PRIVATE #{c}')
+        elif c:
+            out.append(f'c={format_id(c)}')
+
+        if isinstance(u, discord.Member) or isinstance(u, discord.User):
+            out.append(f'({u})')
+        elif u:
+            out.append(f'u={format_id(u)}')
+
+        if isinstance(m, discord.Message):
+            text = ' '.join(m.content.split())
+            if len(text) > 20:
+                text = text[:20].strip() + ' ...'
+            if out: out[-1] = out[-1] + ':'
+            out.append(f'"{text}"')
+        elif m:
+            out.append(f'm={format_id(m)}')
+
+        return ' '.join(out) if out else '(None)'
+
+
+def format_id(id):
```

Comment:
Any reason why this is not in the class, but is in this file? It could go into a snowflake_util.py file, or become a helper method on the class if you think it won't need to be shared outside of this file.